server/form-pages/apply/health-needs/health-needs/learningDifficultiesDetails.test.ts

```diff
@@ -0,0 +1,135 @@
+import LearningDifficultiesDetails, { LearningDifficultiesDetailsBody } from './learningDifficultiesDetails'
+
+describe('LearningDifficultiesDetails', () => {
+  const application = applicationFactory.build({ person: personFactory.build({ name: 'Roger Smith' }) })
+
+  describe('title', () => {
+    it('personalises the page title', () => {
+      const page = new LearningDifficultiesDetails({}, application)
+
+      expect(page.title).toEqual('Add learning difficulties and neurodiversity needs details for Roger Smith')
+    })
+  })
+
+  itShouldHaveNextValue(new LearningDifficultiesDetails({}, application), 'brain-injury')
+  itShouldHavePreviousValue(new LearningDifficultiesDetails({}, application), 'learning-difficulties')
+
+  describe('errors', () => {
+    describe('when top-level questions are unanswered', () => {
+      const page = new LearningDifficultiesDetails({}, application)
+
+      it('includes a validation error for _learningNeedsDetail_', () => {
+        expect(page.errors()).toHaveProperty('learningNeedsDetail', 'Enter details of their needs')
+      })
+
+      it('includes a validation error for _needsSupport_', () => {
+        expect(page.errors()).toHaveProperty('needsSupport', 'Select yes if they need any support')
+      })
+
+      it('includes a validation error for _receivesTreatment_', () => {
+        expect(page.errors()).toHaveProperty('receivesTreatment', 'Select yes if they receive any treatment')
+      })
+
+      it('includes a validation error for _isVulnerable_', () => {
+        expect(page.errors()).toHaveProperty('isVulnerable', 'Select yes if they are vulnerable')
+      })
+    })
+
+    describe('when _needsSupport_ is YES', () => {
+      const page = new LearningDifficultiesDetails({ needsSupport: 'yes' }, application)
+
+      describe('and _interactionDetail_ is UNANSWERED', () => {
+        it('includes a validation error for _supportDetail_', () => {
+          expect(page.errors()).toHaveProperty('supportDetail', 'Enter the type of support needed')
+        })
+      })
+    })
+
+    describe('when _receivesTreatment_ is YES', () => {
+      const page = new LearningDifficultiesDetails({ receivesTreatment: 'yes' }, application)
+
+      describe('and _treatmentDetail_ is UNANSWERED', () => {
+        it('includes a validation error for _treatmentDetail_', () => {
+          expect(page.errors()).toHaveProperty('treatmentDetail', 'Enter details about their treatment')
+        })
+      })
+    })
+
+    describe('when _isVulnerable_ is YES', () => {
+      const page = new LearningDifficultiesDetails({ isVulnerable: 'yes' }, application)
+
+      describe('and _vulnerabilityDetail_ is UNANSWERED', () => {
+        it('includes a validation error for _vulnerabilityDetail_', () => {
+          expect(page.errors()).toHaveProperty('vulnerabilityDetail', 'Enter how they are vulnerable')
+        })
+      })
+    })
```

Comment:
I don't see a benefit in nesting the `describe`s when there's only one child describe in each. Again, all these tests could be put in a single it.each(), passing the key and error message to each.

Reply:
personally I think there is a benefit in that it separates out the context, both in the test and in the test logs.